worker/dependency_manager.py

```diff
@@ -0,0 +1,104 @@
+import threading
+
+
+class DependencyManager(object):
+    """
+    Manages the dependencies available on the worker, and ensures that no two
+    runs download the same dependency at the same time.
+    """
+    # TODO(klopyrev): Delete dependencies if they are using too much disk space.
+    # TODO(klopyrev): Allow downloading only specific subpaths within a
+    #                 dependency.
+    def __init__(self):
+        self._lock = threading.Lock()
+        self._dependencies = {}
+
+    def dependencies(self):
+        """
+        Returns the UUIDs of all the dependencies stored on the worker.
+        """
+        with self._lock:
+            return self._dependencies.keys()
+
+    def add_dependee(self, uuid, dependee_uuid):
```

Comment:
I think 'dependee' is confusing (and I think it actually means the opposite of what you want).  Use 'child' instead.